docs/decisions/0049-agents-assistantsV2.md

```diff
@@ -0,0 +1,183 @@
+
+> Note: Manual function calling isn't currently supported for `OpenAIAssistantAgent` or `AgentChat`  and is planned to be addressed as an enhancement.  When this supported is introduced, `AssistantToolCallBehavior` will determine the function calling behavior (also in alignment with the `ToolCallBehavior` associated with _AI Connectors_).
+
+**Alternative (Future?)**
+
+A pending change has been authored that introduces `FunctionChoiceBehavior` as a property of the base / abstract `PromptExecutionSettings`.  Once realized, it may make sense to evaluate integrating this pattern for `OpenAIAssistantAgent`.  This may also imply in inheritance relationship of `PromptExecutionSettings` for both `OpenAIAssistantExecutionSettings` and `OpenAIAssistantInvocationSettings` (next section).
+
+
+<p align="center">
+<kbd><img src="diagrams/assistant-definition.png"  style="width: 500pt;"></kbd>
+</p>
+
+
+#### Assistant Invocation Settings
+
+When invoking an `OpenAIAssistantAgent` directly (no-chat), settings that only apply to a discrete run may be specified.  These settings are defined as `OpenAIAssistantInvocationSettings` and ovetake precedence over any corresponding assistant or thread settings.
+
+> Note: These settings are also impacted by the `ToolCallBehavior` / `FunctionChoiceBehavior` quadary.
+
+<p align="center">
+<kbd><img src="diagrams/assistant-invocationsettings.png" style="width: 370pt;"></kbd>
+</p>
+
+
+#### Thread Creation Settings
+
+When invoking an `OpenAIAssistantAgent` directly (no-chat), a thread must be explicitly managed.  When doing so, thread specific settings may be specified.  These settings are defined as `OpenAIThreadCreationSettings` and take precedence over any corresponding assistant settings.
+
+<p align="center">
+<kbd><img src="diagrams/assistant-threadcreationsettings.png" style="width: 132pt;"></kbd>
+</p>
+
+
+#### Service Configuration
+
+The `OpenAIServiceConfiguration` defines how to connect to a specific remote service, whether it be OpenAI, Azure, or proxy.  This eliminates the need to define multiple overloads for each call site that results in a connection to the remote API service (i.e. create a _client)_.
+
+> Note: This was previously named `OpenAIAssistantConfiguration`, but is not necessarily assistant specific.
+
+<p align="center">
+<kbd><img src="diagrams/assistant-serviceconfig.png"  style="width: 400pt;"></kbd>
+</p>
+
+
+### Vector Store Support
+
+_Vector Store_ support is required in order to enable usage of the `file-search` tool.  
+
+In alignment with V2 streaming of the `FileClient`, the caller may also directly target `VectorStoreClient` from the _OpenAI SDK_.
+
+
+### Class Inventory
+This section provides an overview / inventory of all the public surface area described in this ADR.
+
+|Class Name|Description|
```

Comment:
Is this table missing the `OpenAIThreadCreationSettings` class?

Reply:
Indeed, updated.